src/components/search/TypeFilter.jsx

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Stanford University see LICENSE for license
+import React, { useState, useEffect } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { fetchSinopiaSearchResults } from 'actionCreators/search'
+import _ from 'lodash'
+
+const TypeFilter = () => {
+  const query = useSelector(state => state.selectorReducer.search.query)
+  const searchOptions = useSelector(state => state.selectorReducer.search.options)
+  const typeFacetResults = useSelector(state => state.selectorReducer.search.facetResults?.types)
+
+  const dispatch = useDispatch()
+  const handleFilter = () => {
+    dispatch(fetchSinopiaSearchResults(query, { ...searchOptions, startOfRange: 0, typeFilter: selectedTypeFilters }))
+    setTypeFilterShowDropdown(false)
+  }
+
+  const clearFilter = () => {
+    dispatch(fetchSinopiaSearchResults(query, { ...searchOptions, startOfRange: 0, typeFilter: undefined }))
+    setTypeFilterShowDropdown(false)
+  }
+
+  const handleOnly = (type) => {
+    setSelectedTypeFilters([type])
+  }
+
+  const [typeFilterShowDropdown, setTypeFilterShowDropdown] = useState(false)
```

Comment:
Not a blocker but should these state functions be moved to line 11 to make it more consistent with other components?